cmd/edgectl/aes_install.go

```diff
@@ -119,65 +119,107 @@ func aesInstall(cmd *cobra.Command, args []string) error {
 	return nil
 }
 
+func getManifest(url string) (string, error) {
+	res, err := http.Get(url)
+	if err != nil {
+		return "", err
+	}
+	bodyBytes, err := ioutil.ReadAll(res.Body)
+	res.Body.Close()
+	if err != nil {
+		return "", err
+	}
+	if res.StatusCode != 200 {
+		return "", errors.Errorf("Bad status: %d", res.StatusCode)
+	}
+	return string(bodyBytes), nil
+}
+
 func (i *Installer) Perform(kcontext string) error {
+	// Start
 	i.Report("install")
 
+	// Download AES manifests
+	crdManifests, err := getManifest("https://www.getambassador.io/yaml/aes-crds.yaml")
+	if err != nil {
+		i.Report("fail_no_internet", ScoutMeta{"err", err.Error()})
+		return errors.Wrap(err, "download AES CRD manifests")
+	}
+	aesManifests, err := getManifest("https://www.getambassador.io/yaml/aes.yaml")
+	if err != nil {
+		i.Report("fail_no_internet", ScoutMeta{"err", err.Error()})
+		return errors.Wrap(err, "download AES manifests")
+	}
+
+	// Figure out what version of AES is being installed
+	// TODO: Parse the manifests and build objects
+	// TODO: Extract version info from the Deployment object
+	// TODO? Set label(s) on the to indicate this installation was performed by the installer
+	aesVersionRE := regexp.MustCompile("image: quay[.]io/datawire/aes:([[:^space:]]+)[[:space:]]")
+	matches := aesVersionRE.FindStringSubmatch(aesManifests)
+	if len(matches) != 2 {
+		i.log.Printf("matches is %+v", matches)
+		return errors.Errorf("Failed to parse downloaded manifests. Is there a proxy server interfering with HTTP downloads?")
+	}
+	aesVersion := matches[1]
+	i.SetMetadatum("AES version being installed", "aes_version", aesVersion)
+
 	// Display version information
-	// TODO: This displays the version of Edge Control, not the image version in
-	// the manifests being downloaded. We should figure out what to do if those
-	// don't match up.
-	// 1. Allow an old Edge Control to install a newer AES
-	// 2. Insist that Edge Control be the same version as the AES it's installing
-	// 3. Something else?
-	// Note that the second option will allow us to make the install process
-	// more complicated in the future without having to subject our users to
-	// increasing complexity, assuming this approach to installation gains
-	// traction.
-	i.show.Printf("-> Installing the Ambassador Edge Stack %s\n", Version)
+	i.show.Printf("-> Installing the Ambassador Edge Stack %s\n", aesVersion)
 
 	// Attempt to talk to the specified cluster
+	// TODO: Figure out cluster info to be passed along
 	i.kubeinfo = k8s.NewKubeInfo("", kcontext, "")
-
-	if err := i.ShowKubectl("cluster-info", "cluster-info"); err != nil {
+	if err := i.ShowKubectl("cluster-info", "", "cluster-info"); err != nil {
+		i.Report("fail_no_cluster")
 		return err
 	}
+	i.SetMetadatum("Cluster info", "cluster_info", "FIXME")
```

Comment:
Probably needs to rebase this change with shared/2ksec

Reply:
Huh? GitHub marks this as outdated. It's not in my copy either.